src/User32.Desktop/User32+DESKTOP_ACCESS_MASK.cs

```diff
@@ -0,0 +1,268 @@
+            /// Bits 16-23
+            /// </summary>
+            private const uint StandardRightsMask = 0x00ff0000;
+
+            /// <summary>
+            /// Bits 0-15
+            /// </summary>
+            private const uint SpecificRightsMask = 0x0000ffff;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DESKTOP_ACCESS_MASK"/> struct.
+            /// </summary>
+            /// <param name="value">The value for the <see cref="DESKTOP_ACCESS_MASK"/>.</param>
+            public DESKTOP_ACCESS_MASK(uint value)
+            {
+                this.Value = value;
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DESKTOP_ACCESS_MASK"/> struct.
+            /// </summary>
+            /// <param name="value">The reference <see cref="Kernel32.ACCESS_MASK"/> for the <see cref="DESKTOP_ACCESS_MASK"/>.</param>
+            public DESKTOP_ACCESS_MASK(Kernel32.ACCESS_MASK value)
+            {
+                this.Value = value;
+            }
+
+            /// <summary>
+            /// The following are the generic access rights for a desktop object contained in the interactive window station of the user's logon session
+            /// </summary>
+            [Flags]
+            public enum GenericRight : uint
+            {
+                GENERIC_READ = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_READ |
+                    SpecificRight.DESKTOP_ENUMERATE |
+                    SpecificRight.DESKTOP_READOBJECTS,
+                GENERIC_WRITE = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_WRITE |
+                    SpecificRight.DESKTOP_CREATEMENU |
+                    SpecificRight.DESKTOP_CREATEWINDOW |
+                    SpecificRight.DESKTOP_HOOKCONTROL |
+                    SpecificRight.DESKTOP_JOURNALPLAYBACK |
+                    SpecificRight.DESKTOP_JOURNALRECORD |
+                    SpecificRight.DESKTOP_WRITEOBJECTS,
+                GENERIC_EXECUTE = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_EXECUTE |
+                    SpecificRight.DESKTOP_SWITCHDESKTOP,
+                GENERIC_ALL = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_REQUIRED |
+                    SpecificRight.DESKTOP_CREATEMENU |
+                    SpecificRight.DESKTOP_CREATEWINDOW |
+                    SpecificRight.DESKTOP_ENUMERATE |
+                    SpecificRight.DESKTOP_HOOKCONTROL |
+                    SpecificRight.DESKTOP_JOURNALPLAYBACK |
+                    SpecificRight.DESKTOP_JOURNALRECORD |
+                    SpecificRight.DESKTOP_READOBJECTS |
+                    SpecificRight.DESKTOP_SWITCHDESKTOP |
+                    SpecificRight.DESKTOP_WRITEOBJECTS
+            }
+
+            /// <summary>
+            /// Contains the access mask specific to the Desktop associated with the mask.
+            /// </summary>
+            [Flags]
+            public enum SpecificRight : uint
```

Comment:
And here for the SpecificRight implementation for that kind of objects. This we might see a lot when we map security for devices, drivers, windows objects (internal and user ones).